x-pack/plugins/alerting/server/task_runner/inject_action_params.ts

```diff
@@ -9,25 +9,28 @@ import { i18n } from '@kbn/i18n';
 import { AlertActionParams } from '../types';
 
 export interface InjectActionParamsOpts {
-  alertId: string;
+  ruleId: string;
+  spaceId: string;
   actionTypeId: string;
   actionParams: AlertActionParams;
 }
 
 export function injectActionParams({
-  alertId,
+  ruleId,
+  spaceId,
   actionTypeId,
   actionParams,
 }: InjectActionParamsOpts) {
   // Inject kibanaFooterLink if action type is email. This is used by the email action type
   // to inject a "View alert in Kibana" with a URL in the email's footer.
   if (actionTypeId === '.email') {
+    const spacePrefix = spaceId.length > 0 && spaceId !== 'default' ? `/s/${spaceId}` : '';
```

Comment:
Is the `spaceId.length > 0` check necessary? Is it possible to not be in a space, or in a space without a name?

Reply:
I think it's not possible? I wanted to cover all my bases am happy to remove.

Reply:
I'd defer to your judgment here, but I wasn't sure if it was possible or not

Reply:
The only thing I can think of is testing the behaviour when the spaces plugin is disabled `xpack.spaces.enabled: false`.

Reply:
Good call @mikecote. `spaceId` is `undefined` in this scenario 😅 . Updated to handle that here https://github.com/elastic/kibana/pull/100734/commits/64414fccd52cf604030dbcab39dfc0c9ef18cca4